### PR TITLE
Feature/1797 register node return status

### LINKED ISSUE #1797: pf::node::node_register | Refactor to add return code + status code/message
It will be nice to have the status page url once you have reached the maximum node that you can register

### PR DESCRIPTION
**Pull request by dwlfrth**
# Description

pf::node::node_register is able to return status message to inform end users on failure per example 
# Impacts

There shouldn't be any impact since we basically return a string along with the return code (0,1)
Should just make sure all the caller are handling the return code as they should (curl ?)
pf::node::node_register flow
- registration
- auto-registration
- wispr
- node import
- administration interface (user / nodes)
- portal
- vlan_filters (action = register_node)
- wmi (action = dynamic_register_node)

# Delete branch after merge

YES
# NEWS file entries
## Enhancements
- Portal is giving more indication on node registration error
## Bug Fixes
- pf::node::node_register | Refactor to add return code + status code/message (#1797)

### CODE REVIEW

**Comment by dwlfrth** [comment]
Please review

**Comment by julsemaan** [comment]
I'm not sure this really fixes #1797 

It talks about showing the status page in the body but about reworking the node_register in the title.

Something is up...

**Comment by dwlfrth** [comment]
It partially fixes it in the sense that node_register will return a message to the portal.
Second part (subject of the issue) is more a customization and may be a good thing to be integrated in PacketFence codebase... having the message to display a link to the "status" page.

**Comment by dwlfrth** [comment]
Bump

File: lib/pf/node.pm [hunk 1]
```diff
@@ -976,8 +978,9 @@ sub node_register {
     else {
     # do not check for max_node if it's for auto-register
         if ( is_max_reg_nodes_reached($mac, $pid, $info{'category'}, $info{'category_id'}) ) {
-            $logger->error( "max nodes per pid met or exceeded - registration of $mac to $pid failed" );
-            return (0);
+            $status_msg = "max nodes per pid met or exceeded";
+            $logger->error( "$status_msg - registration of $mac to $pid failed" );
+            return (0, $status_msg);
```
Comment (julsemaan) [inline lib/pf/node.pm]: Use the FALSE Luke

**Comment by julsemaan** [comment]
Just fix the constant and we're good to merge

**Comment by dwlfrth** [comment]
Sup !

**Comment by julsemaan** [comment]
https://www.youtube.com/watch?v=lgJsNW_zNUI&t=21

**Comment by extrafu** [comment]
Bump

**Pull request merged**